backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/services/RentalService.java

```diff
@@ -63,22 +59,20 @@ public Rental getRentalById(Long id) {
 	}
 
 	public Rental createRental(RentalDTO rentalDTO) {
-		//TODO: ADD USER AUTHENTICATION
+		User user = userService.getLoggedInUser();
+		if(user == null) {
+			throw new AuthorizationException();
+		}
 		ModelMapper mapper = new ModelMapper();
 		Rental rental = mapper.map(rentalDTO, Rental.class);
+		rental.setUser(user);
 		List<Rental> rentals = getRentalsByBookId(rentalDTO.getBookId());
 		if (rentals != null && !rentals.isEmpty()) {
 			throw new BookRentedException(rentalDTO.getBookId());
 		}
-
-		Long bookId = rentalDTO.getBookId();
-		Book borrowedBook = bookService.getBookById(bookId);
-		BookStatus statusBorrowed = bookStatusService.getBookStatusById(2L);
-		borrowedBook.setStatus(statusBorrowed);
-		borrowedBook.setAvailableDate(LocalDate.now().plusMonths(1));
-		rental.setBook(borrowedBook);
-
-		rental.setUser(userService.getUserById(rentalDTO.getUserId()));
+		Book borrowedBook = bookService.getBookById(rentalDTO.getBookId());
+		rental.setBook(bookService.changeStatus(borrowedBook, 2L));
```

Comment:
changeStatus to 2L ? use some enum or anything that one can actually read. Maybe instead of setting 2L just change create several methods: 'changeStatusToRented', or 'setStatusReserved' ? (ps, why long?)